core/vm/contracts.go

```diff
@@ -251,23 +296,25 @@ func (c *bigModExp) RequiredGas(input []byte) uint64 {
 		adjExpLen.Mul(big8, adjExpLen)
 	}
 	adjExpLen.Add(adjExpLen, big.NewInt(int64(msb)))
```

Comment:
The new EIP at least has some funky part:

```
def calculate_iteration_count(exponent_length, exponent):
    iteration_count = 0
    if exponent_length <= 32 and exponent == 0: iteration_count = 0
    elif exponent_length <= 32: iteration_count = exponent.bit_length() - 1
    elif exponent_length > 32: iteration_count = (8 * (exponent_length - 32)) + ((exponent & (2**256 - 1)).bit_length() - 1)
    return max(iteration_count, 1)
```

Namely, in the 3rd clause, it has `((exponent & (2**256 - 1)).bit_length()` vs `exponent.bit_length()` in the second. The code however treats both cases as `msb`. Is this correct?

Alternatively, `& (2**256 - 1)` seems to mean "truncate to 32 bytes". But when can it be larger than 32 bytes? Can it? If not, I think we need to clear up the spec as this is super weird.

Reply:
I think it means truncate. I'm not sure -- that is old code..

Reply:
I hadn't seen that final version of the EIP -- however, the EIP consists of three separate changes to the previous version. That's what's implemented.